test/unit/domain/authorizations.test.ts

```diff
@@ -0,0 +1,282 @@
+      async (): Promise<void> => {
+        const payload: AuthPayload = {
+          consentId: '1223abcd',
+          sourceAccountId: '2222-322d-d2k2',
+          status: 'PENDING',
+          challenge: 'dddw7hwuehfuhnd8jd',
+          value: 'dwuduwd&e2idjoj0w'
+        }
+
+        const consentScopes: Scope[] = [
+          {
+            id: 1,
+            consentId: payload.consentId,
+            action: 'account.transfer',
+            accountId: '3332-edds-2332'
+          },
+          {
+            id: 2,
+            consentId: payload.consentId,
+            action: 'account.balance',
+            accountId: payload.sourceAccountId
+          },
+          {
+            id: 3,
+            consentId: payload.consentId,
+            action: 'account.billpayment',
+            accountId: '2020-20sj-nsj2'
+          }
+        ]
+
+        const scopeMatch = hasMatchingScopeForPayload(consentScopes, payload)
+
+        expect(scopeMatch).toEqual(true)
+      })
+
+    it('returns false if the payload scope does not match any consent scopes',
+      async (): Promise<void> => {
+        const payload: AuthPayload = {
+          consentId: '1223abcd',
+          sourceAccountId: '2222-322d-d2k2',
+          status: 'PENDING',
+          challenge: 'dddw7hwuehfuhnd8jd',
+          value: 'dwuduwd&e2idjoj0w'
+        }
+
+        const consentScopes: Scope[] = [
+          {
+            id: 1,
+            consentId: payload.consentId,
+            action: 'account.transfer',
+            accountId: '3332-edds-2332'
+          },
+          {
+            id: 2,
+            consentId: payload.consentId,
+            action: 'account.balance',
+            accountId: '3332-2dcx-1020'
+          },
+          {
+            id: 3,
+            consentId: payload.consentId,
+            action: 'account.billpayment',
+            accountId: '2020-20sj-nsj2'
+          }
+        ]
+
+        const scopeMatch = hasMatchingScopeForPayload(consentScopes, payload)
+
+        expect(scopeMatch).toEqual(false)
+      })
+  })
+
+  describe('putErrorRequest', (): void => {
+    let mockErrorMethod: jest.SpyInstance
+    let request: Request
+
+    beforeAll((): void => {
+      mockErrorMethod = jest.spyOn(
+        thirdPartyRequest,
+        'putThirdpartyRequestsTransactionsAuthorizationsError'
+      )
+
+      // @ts-ignore
+      request = {
+        headers: {
+          'fspiop-source': 'switch'
+        },
+        params: {
+          id: '1234'
+        },
+        payload: {
+          consentId: '1234',
+          sourceAccountId: 'pisp-2343-f223',
+          status: 'PENDING',
+          challenge: 'QuoteResponse Object JSON string',
+          value: 'YjYyODNkOWUwZjUxNzOThmMjllYjE2Yg=='
+        }
+      }
+    })
+
+    it('calls the thirdPartyRequest error method with correct paramaeters',
+      async (): Promise<void> => {
+        mockErrorMethod.mockResolvedValue(true)
```

Comment:
this is invalid mock! `putErrorRequest` returns `Promise<void>` not `true`

Reply:
mockErrorMethod -> mockSDKErrorMethod
true -> {} as GenericResponse